File: Demo/Demo/Shapes/Circle.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Demo.Shapes
{
	class Circle : Shape
	{
		private Point center;

		public Circle(Point center, int radius)
		{
			this.center = center;
			this.Radius = radius;
		}

		public int Radius { get; private set; }

		public override Point Center
		{
			get { return center; }
		}

		public override void Print(TextWriter output)
		{
			output.WriteLine($"{this.Color} circle at {this.Center} of radius {this.Radius}");
		}
	}
}

File: Demo/DemoVisualBasic/Shapes/Circle.vb

Imports System.Collections.Generic
Imports System.IO
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks

Namespace Demo.Shapes
	Class Circle
		Inherits Shape
		Private _center As Point

		Public Sub New(center As Point, radius As Integer)
			Me._center = center
			Me.Radius = radius
		End Sub

		Public ReadOnly Property Radius As Integer

		Public Overrides ReadOnly Property Center() As Point
			Get
				Return _center
			End Get
		End Property

		Public Overrides Sub Print(output As TextWriter)
			output.WriteLine($"{Me.Color} circle at {Me.Center} of radius {Me.Radius}")
		End Sub
	End Class
End Namespace
